src/radio/radio-controller.ts

```diff
@@ -0,0 +1,233 @@
+export class UDPScanConfig {
+  broadcastAddress: string;
+  broadcastPort: number;
+  listenPort: number;
+  discoveryPacket: string;
+  /**
+   *
+   * @param broadcastAddress  The destination UDP broadcast address.
+   * @param broadcastPort  The destination UDP broadcast port.
+   * @param listenport  The listen port for the UDP response.
+   * @param discoveryPacket  The payload to send in the UDP broadcast.
+   * @returns  A new `UDPScanConfig` instance.
+   */
+  constructor(
+    broadcastAddress: string,
+    broadcastPort: number,
+    listenport: number,
+    discoveryPacket: string
+  ) {
+    this.broadcastAddress = broadcastAddress;
+    this.broadcastPort = broadcastPort;
+    this.listenPort = listenport;
+    this.discoveryPacket = discoveryPacket;
+  }
+}
+
+/**
+ * A class to contain the information from a UDP scan.
+ */
+export interface UDPScanResults {
+  buffer: Buffer;
+  address: string;
+}
+
+/**
+ * A class to contain all Node radio functionality.
+ */
+export class RadioController {
+  /**
+   * A helper function to create timeout promises.
+   * @returns  A new Promise that resolves after RADIO_TIMEOUS milliseconds.
+   */
+  private createTimeoutPromise(): Promise<void> {
+    return new Promise(resolve => {
+      setTimeout(resolve, RADIO_TIMEOUT);
+    });
+  }
+
+  /**
+   * Performs a UDP scan according to a `UDPScanConfig`.
+   * @param udpScanConfig  A scan configuration containing UDP parameters.
+   * @return  A promise that resolves to the determined `UDPScanResults`
+   */
+  public async udpScan(udpScanConfig: UDPScanConfig): Promise<UDPScanResults> {
+    // Open a UDP socket.
+    const socket = dgram.createSocket('udp4');
+    const discoveryBuffer = new Promise<UDPScanResults>(resolve => {
+      // Resolve promise when a broadcast buffer is recieved.
+      socket.on('message', (msg, rinfo) => {
+        // Close the socket.
+        socket.close();
+        resolve({buffer: msg, address: rinfo.address});
+      });
+
+      // Enable UDP broadcast.
+      socket.on('listening', () => {
+        socket.setBroadcast(true);
+      });
+      socket.bind(udpScanConfig.listenPort);
+
+      // Decode the discovery packet and send it.
+      const payload = Buffer.from(udpScanConfig.discoveryPacket, 'hex');
+      socket.send(
+        payload,
+        udpScanConfig.broadcastPort,
+        udpScanConfig.broadcastAddress,
+        error => {
+          if (error !== null) {
+            throw new Error(
+              'Failed to send UDP discovery packet:' + error.message
+            );
+          }
+          console.log('Sent UDP discovery packet: ', payload);
+        }
+      );
+    });
+    // Timeout if there hasn't been a response.
+    return Promise.race([
+      discoveryBuffer,
+      this.createTimeoutPromise().then(() => {
+        // Close the socket if timed out
+        socket.close();
+        throw new Error(
+          'UDP scan timed out after ' + RADIO_TIMEOUT.toString() + 'ms.'
```

Comment:
You can use string templating here, and you shouldn't need to cast the number explicitly.

`... ${RADIO_TIMEOUT} ms`.

Reply:
Will fix in #48